patch/ladspa-sdk/001-fix-memleak.patch

```diff
@@ -0,0 +1,11 @@
+--- src/search.c.orig	2008-11-07 00:38:18.000000000 +0100
++++ src/search.c	2008-11-07 00:50:51.000000000 +0100
+@@ -83,6 +83,8 @@
+ 	dlclose(pcFilename);
+ 	free(pcFilename);
```

Comment:
I didn't see that file in the manifest.

Reply:
Do you mean that the patch should be removed as it's no longer used? We are using `- "shared-modules/linux-audio/ladspa.json"`, but I don't think patches are automatically applied to shared modules so the patch can probably go.

Reply:
there is no magic, if it's not listed, it's not used.

On the other hand, if that's really a necessary patch I'd be happy to add it to the shared module that that other packages can benefit from it.

Reply:
@AsavarTzeth could you comment if this `001-fix-memleak.patch` patch is still useful for EasyEffects, and/or if it should be contributed to shared-modules?

This patch was originally here: https://github.com/flathub/com.github.wwmm.pulseeffects/blob/master/patch/ladspa-sdk/001-fix-memleak.patch

Reply:
No response, they don't seem to be active :(

[The patch was added to PulseEffects in 2017](https://github.com/flathub/com.github.wwmm.pulseeffects/commit/bc8dfacf1ad06a3d188c9a1a53ab808da4a0eeb9), and I haven't found any explanations for it. And the date is from 2008...

So I just looked at the [latest ladspa sdk (1.15) source](https://www.ladspa.org/download/index.html) instead.

<details><summary>/src/source.c</summary><br/>

```c
/* search.c

   Free software by Richard W.E. Furse. Do with as you will. No
   warranty. */

/*****************************************************************************/

#include <dirent.h>
#include <dlfcn.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <sys/types.h>
#include <unistd.h>

/*****************************************************************************/

#include "ladspa.h"
#include "utils.h"

/*****************************************************************************/

/* Search just the one directory. */
static void
LADSPADirectoryPluginSearch
(const char * pcDirectory, 
 LADSPAPluginSearchCallbackFunction fCallbackFunction) {

  char * pcFilename;
  DIR * psDirectory;
  LADSPA_Descriptor_Function fDescriptorFunction;
  long lDirLength;
  long iNeedSlash;
  struct dirent * psDirectoryEntry;
  void * pvPluginHandle;

  lDirLength = strlen(pcDirectory);
  if (!lDirLength)
    return;
  if (pcDirectory[lDirLength - 1] == '/')
    iNeedSlash = 0;
  else
    iNeedSlash = 1;

  psDirectory = opendir(pcDirectory);
  if (!psDirectory)
    return;

  while (1) {

    psDirectoryEntry = readdir(psDirectory);
    if (!psDirectoryEntry) {
      closedir(psDirectory);
      return;
    }

    pcFilename = malloc(lDirLength
			+ strlen(psDirectoryEntry->d_name)
			+ 1 + iNeedSlash);
    strcpy(pcFilename, pcDirectory);
    if (iNeedSlash)
      strcat(pcFilename, "/");
    strcat(pcFilename, psDirectoryEntry->d_name);
    
    pvPluginHandle = dlopen(pcFilename, RTLD_LAZY);
    if (pvPluginHandle) {
      /* This is a file and the file is a shared library! */

      dlerror();
      fDescriptorFunction
	= (LADSPA_Descriptor_Function)dlsym(pvPluginHandle,
					    "ladspa_descriptor");
      if (dlerror() == NULL && fDescriptorFunction) {
	/* We've successfully found a ladspa_descriptor function. Pass
           it to the callback function. */
	fCallbackFunction(pcFilename,
			  pvPluginHandle,
			  fDescriptorFunction);
	free(pcFilename);
      }
      else {
	/* It was a library, but not a LADSPA one. Unload it. */
	dlclose(pvPluginHandle);
	free(pcFilename);
      }
                          
    }
  }
}

/*****************************************************************************/

void 
LADSPAPluginSearch(LADSPAPluginSearchCallbackFunction fCallbackFunction) {

  char * pcBuffer;
  const char * pcEnd;
  const char * pcLADSPAPath;
  const char * pcStart;

  pcLADSPAPath = getenv("LADSPA_PATH");
  if (!pcLADSPAPath) {
    fprintf(stderr,
	    "Warning: You do not have a LADSPA_PATH "
	    "environment variable set. Defaulting to "
            EXPAND_AND_STRINGIFY(DEFAULT_LADSPA_PATH)
            ".\n");
    pcLADSPAPath = EXPAND_AND_STRINGIFY(DEFAULT_LADSPA_PATH);
  }
  
  pcStart = pcLADSPAPath;
  while (*pcStart != '\0') {
    pcEnd = pcStart;
    while (*pcEnd != ':' && *pcEnd != '\0')
      pcEnd++;
    
    pcBuffer = malloc(1 + pcEnd - pcStart);
    if (pcEnd > pcStart)
      strncpy(pcBuffer, pcStart, pcEnd - pcStart);
    pcBuffer[pcEnd - pcStart] = '\0';
    
    LADSPADirectoryPluginSearch(pcBuffer, fCallbackFunction);
    free(pcBuffer);

    pcStart = pcEnd;
    if (*pcStart == ':')
      pcStart++;
  }
}

/*****************************************************************************/

/* EOF */
```
</details>

I am unsure if the patch applies here, since the code seems to have changed. Furthermore, I can't find older sdk revisions since the 1.13 ladspa sdk url is a 404, and I can't find a repo for it. 

I didn't see anything in [recent changelogs](https://www.ladspa.org/ladspa_sdk/changes.html) about memleak fixes though, so maybe this is worth investigating.

Should I open an issue to discuss this in shared-modules or I am looking too far into this?